quaddtype/numpy_quaddtype/src/ops.hpp

```diff
@@ -1238,3 +1238,83 @@ ld_greaterequal(const long double *a, const long double *b)
 {
     return *a >= *b;
 }
+
+// Logical operations
+
+// Helper function to check if a Sleef_quad value is non-zero (truthy)
+static inline npy_bool
+quad_is_nonzero(const Sleef_quad *a)
+{
+    // A value is falsy if it's exactly zero (positive or negative)
+    // NaN and inf are truthy
+    npy_bool is_zero = Sleef_icmpeqq1_purec(*a, QUAD_ZERO);
```

Comment:
Why are we using the purec version here, I think we've never used the specific prefixes?

Reply:
Ah yes they are pure c implementation is just a specialization and "no-suffix" one can sometimes map to FMA versions if hardware supports else pure-c.

Let me change it to keep it consistent